### DISCUSSION TITLE
How to autostart public web player? (Useful for smartphones)

### DISCUSSION THREAD

**Discussion by Moonbase59** [AzuraCast/AzuraCast#6770] [Q&A] [answered]
I couldn’t find anything in the docs. So **_can_ we autostart the public web player somehow**, maybe by some URL parameter like `?autostart=true`?

Rationale: If you use the same popup window for the web player and have some means to select different stations on your website, on each change we get a new player in the same window, which is not playing by default.

Now on small mobile devices like smartphones, the tab/window might be hidden below other open pages in the browser, and users complain that they always have to navigate, search the tab, and start the player manually again. Which _can_ be frustrating if you have many tabs open on a small device.

**Comment by Moonbase59** [comment]
I played around a little and found `?autoplay=true` seems to do it.

Is that official and documented somewhere?

**Reply by BusterNeece** [reply, marked as answer]
@Moonbase59 If you look through old GitHub issues/discussions you might find the answer, but to repeat it here: it's not officially documented, nor is it recommended, because a fair number of modern browsers have taken active countermeasures _against_ autoplaying audio/video when a page loads. The number is enough that the experience isn't a consistent one with that flag enabled, so it's probably not worth using in most situations.

**Reply by marekdziem** [reply]
> I played around a little and found `?autoplay=true` seems to do it.
> 
> Is that official and documented somewhere?

ah thank you ! brilliant!

**Comment by Moonbase59** [comment]
It’s really sad that some freakin’ idiots always destroy everything on the Internet that could otherwise be useful features. It's unbelievable what economic damage this causes, if you look at the big picture.

Anyway, it’s a good feature to have and works even with browser restrictions if done rightly. So please don’t remove the possibility.

I, for instance, open the AzuraCast public player(s) in a (re-used) popup window from an `onclick` event (i.e., a "Play" button) on my website. Since this _is_ an intended user interaction, all browsers I tested allow autoplay on it, and it’s a good way to change streams or stations when users click on the desired "Play" button.

Tested this on current Firefox, Chromium, Chrome and Opera on various Linuxes, and MS Edge on Win10 and Win11 as well as on Android smartphones using Chrome and Firefox.

Works on all of them except Android Firefox which exposes a bug: Despite setting the correct "allow audio autoplay" option, it is ignored: The player starts but the sound is muted. Android Chrome does it correctly. Will have some friends test on iOS devices shortly, I don’t have any Apple stuff.

[Sample video: Firefox on Laptop](https://github.com/AzuraCast/AzuraCast/assets/3706922/732cce96-6d92-49f8-9ee3-9a6e968e6b00)

(Sorry, for some reason GitHub removes the sound, but you can see the player button…)
